### ISSUE TITLE
Drush 11.1.1 is not compatible with Drupal 10.

### ISSUE THREAD

**Issue by sonvir249** [drush-ops/drush#5243]
**Describe the bug**
Drush 11.1 is not compatible with Drupal 10.0.0-beta1.

**To Reproduce**
I used `composer require drush/drush -W`

**Expected behavior**
It should install the Drush 11.1 

**Actual behavior**
Getting below errors
```  Problem 1
    - consolidation/log[1.1.1, 2.0.2, ..., 2.1.0] require psr/log ^1.0 -> found psr/log[1.0.0, ..., 1.1.4] but these were not loaded, likely because it conflicts with another require.
    - consolidation/log 2.1.1 requires psr/log ^1 || ^2 -> found psr/log[1.0.0, ..., 1.1.4, 2.0.0] but these were not loaded, likely because it conflicts with another require.
    - consolidation/robo 3.0.9 requires symfony/process ^4.4.9 || ^5 -> found symfony/process[v4.4.9, ..., v4.4.44, v5.0.0-BETA1, ..., v5.4.11] but these were not loaded, likely because it conflicts with another require.
    - drush/drush[11.1.0, ..., 11.1.1] require consolidation/robo ^3.0.9 || ^4.0.0-alpha1 -> satisfiable by consolidation/robo[3.0.9, 3.0.10].
    - consolidation/robo 3.0.10 requires consolidation/log ^1.1.1 || ^2.0.2 -> satisfiable by consolidation/log[1.1.1, 2.0.2, ..., 2.1.1].
    - Root composer.json requires drush/drush ^11.1 -> satisfiable by drush/drush[11.1.0, 11.1.1].

You can also try re-running composer require with an explicit version constraint, e.g. "composer require drush/drush:*" to figure out if any version is installable, or "composer require drush/drush:^2.1" if you know which you need.
```
**Workaround**
I had to install below libraries manually and then `composer require drush/drush` worked
```
composer require consolidation/robo:4.0.0-alpha1
```

### System Configuration
| Q               | A
| --------------- | ---
| Drush version?  | 11.1
| Drupal version? | 10 10.0.0-beta1
| PHP version     | 8.1
| Composer version | 2.4.2
| DDEV version     | 1.21.1
| OS?             | Mac (M1)

**Additional information**
I am using ddev for Drupal setup.

**Comment by gitressa** [comment]
I can confirm that the error occurs, and installing `consolidation/robo:4.0.0-alpha1` fixes it.

I get an error after running `drush site:install` (known issue) which should get fixed with next release, if it includes #5213.

**Comment by weitzman** [comment]
Drush is already [compatible with consolidation/robo:4.0.0-alpha1](https://packagist.org/packages/drush/drush). Someone needs to debug why composer is apparently unwilling to upgrade it. Also please try to run composer update without having any vendor dir.

**Comment by sonvir249** [comment]
@weitzman we can run the `composer update` if drush is already installed. This issue is coming on fresh drush installtion.

**Comment by greg-1-anderson** [comment]
Oh, looks like I need to tag a stable 4.0.0 of consolidation/robo! Not sure if that's part of the problem, but I'll make a release; I thought I already had.

**Comment by gitressa** [comment]
Thanks @greg-1-anderson, perhaps https://github.com/consolidation/log/pull/32#issuecomment-1221598427 could be dealt with also, to avoid the ghastly error message after installing Drupal 10?

**Comment by greg-1-anderson** [comment]
Looks like @weitzman fixed consolidation/log#32. I released Robo 4.0.0 stable; #5247 should help take care of the problems reported here, I think.

**Comment by weitzman** [comment]
Fixed. Release is coming today, I hope. Thanks all.
